src/debug/netcoredbg/manageddebugger.cpp

```diff
@@ -1274,6 +1274,7 @@ HRESULT ManagedDebugger::RunProcess(std::string fileExec, std::vector<std::strin
                                      &m_processId, &resumeHandle));
 
     IfFailRet(g_dbgshim.RegisterForRuntimeStartup(m_processId, ManagedDebugger::StartupCallback, this, &m_unregisterToken));
+    m_protocol->EmitProcessEvent(m_processId, fileExec);
```

Comment:
I think it's better to move this event into `HRESULT CreateProcess(...)` method.

Reply:
I am not a C++ developer, so the current change is the simplest for me to insert this event. I checked
```
virtual HRESULT STDMETHODCALLTYPE CreateProcess(
            /* [in] */ ICorDebugProcess *pProcess)
```
but `ICorDebugProcess` seems to only support process ID query via `GetID`, but not file name query.

Reply:
@lextm Are you sure that you need `fileExec` here? It seems to me that for .NET this should be an assembly name or something similar. But in your current code you would get a `dotnet` executable.
You can enumerate AppDomains with `ICorDebugProcess::EnumerateAppDomains` (for .NET Core there should be only one domain) and then call `ICorDebugAppDomain::GetName`.

See the docs at:
* `ICorDebugProcess::EnumerateAppDomains` https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugprocess-enumerateappdomains-method
* `ICorDebugAppDomain::GetName` https://docs.microsoft.com/en-us/dotnet/framework/unmanaged-api/debugging/icordebugappdomain-getname-method

Reply:
I am almost sure that `dotnet` should be the information to return, as the IDE/editor expects the actual process information (process file name, and process ID).